test/test_parser.py

```diff
@@ -73,13 +76,20 @@ def testMtlTextureName(self):
         self.assertEqual(self.material1.texture.image_name,
                          prepend_dir('4x4.png'))
 
+
 class TestParserFailure(unittest.TestCase):
 
     def testMissingParseFunction(self):
         "Attempting to parse with a missing parse function should raise an exception."
-        # since no parse functions have been defined, this will always fail
-        self.assertRaises(Exception, pywavefront.parser.Parser, prepend_dir('uv_sphere.obj'))
+        # since no parse functions have been defined, this will always fail in strict mode
+        file_name = 'simple.obj'
+        parser = pywavefront.parser.Parser(prepend_dir(file_name))
+        pywavefront.parser.Parser.strict = True
+        self.assertRaises(AttributeError, parser.read_file, prepend_dir(file_name))
+        pywavefront.parser.Parser.strict = False
```

Comment:
Is this line relevant? You're returning from the method here.

Reply:
`strict` is an attribute on the class itself, so I have to revert it or I break every other test. Should probably use a proper mock, but it might be better to make it a proper instance variable in the future.

Reply:
Did you verify that behavior? This is test fixture and as far as I know, the convention for generic fixtures is that the set up and teardown methods are supposed to run before and after each test respectively. So I assumed a clean object should be created from scratch before every test, although I haven't verified it myself.

Reply:
Confirmed; the idea is to set "your" preferred strictness globally. The decorators for controlling that for tests make sense. Also agreed about deferring that.